documentation/src/main/asciidoc/entity-view/manual/en_US/04_mappings.adoc

```diff
@@ -1337,4 +1337,248 @@ public abstract class CatView {
     }
 }
 ----
-<1> If the unsafe proxy is used, `getAge()` will return the actual value, otherwise will return `null`
\ No newline at end of file
+<1> If the unsafe proxy is used, `getAge()` will return the actual value, otherwise will return `null`
+
+=== Inheritance mapping
+
+Entity views can have an inheritance relationship to subtypes via an inheritance mapping. This relationship allows instances of an entity view subtype to be materialized when a selection predicate,
+that is defined by an inheritance mapping, is satisfied.
+
+The inheritance feature for an entity view is activated by annotating link:{entity_view_jdoc}/persistence/view/EntityViewInheritance.html[`@EntityViewInheritance`] on an entity view.
+By default, all subtypes of the entity view are considered as inheritance subtypes and thus require a so called _inheritance mapping_.
+
+A _inheritance mapping_ is defined by annotating the subtype with link:{entity_view_jdoc}/persistence/view/EntityViewInheritanceMapping.html[`@EntityViewInheritanceMapping`] and defining
+a selection predicate that represents the condition on which is decided whether an instance of that subtype should be constructed. The predicate is a normal link:{core_doc}#expressions[JPQL predicate expression]
+and can refer to all attributes of the mapped entity type.
+
+Consider the following example
+
+[source,java]
+----
+@EntityView(Cat.class)
+@EntityViewInheritance
+public interface BaseCatView {
+    String getName();
+}
+
+@EntityView(Cat.class)
+@EntityViewInheritanceMapping("age < 18")
+public interface YoungCatView extends BaseCatView {
+    @Mapping("mother.name")
+    String getMotherName();
+}
+
+@EntityView(Cat.class)
+@EntityViewInheritanceMapping("age > 18")
+public interface OldCatView extends BaseCatView {
+    @Mapping("kittens.name")
+    List<String> getKittenNames();
+}
+----
+
+When querying for entity views of the type `BaseCatView`, the selection predicates `age < 18` and `age > 18` are _merged_ into a _type discriminator_ expression that returns a type index.
+The type index refers to the entity view type into which a result should be materialized. The resulting JPQL query for such an entity view looks like the following
+
+[source,sql]
+----
+SELECT
+    CASE
+        WHEN age < 18 THEN 1
+        WHEN age > 18 THEN 2
+        ELSE 0
+    END,
+    cat.name,
+    mother_1.name,
+    kittens_1.name
+FROM Cat cat
+LEFT JOIN cat.mother mother_1
+LEFT JOIN cat.kittens kittens_1
+----
+
+The type index 0 refers to the base type `BaseCatView` and in this case instances of `BaseCatView` are materialized when the age of a result is exactly 18.
```

Comment:
The type index 0 refers to the base type `BaseCatView`, hence instances of `BaseCatView` are materialized when the age of a result equals 18.